### PR TITLE
Make sure an agent is running before adding it as authentication method.

### LINKED ISSUE #14196: On Windows IdentitiesOnly with public identity file through SSH agent not authenticating
SFTP access.
It works from the Windows shell, but I get an error with Cyberduck:

```
Host hhh
  HostName hhh.cloudapp.azure.com
  User hhh-admin
  IdentityFile \Users\hhh\.ssh\hhh-admin-key.pub
  IdentitiesOnly yes
```

```
ch.cyberduck.core.AbstractExceptionMappingService - No message for failure net.schmizz.sshj.common.SSHException
ch.cyberduck.core.threading.DefaultFailureDiagnostics - Determine cause for failure BackgroundException{class=class ch.cyberduck.core.exception.BackgroundException, file=null, message='Accesso fallito.', detail='Sconosciuto.', cause='net.schmizz.sshj.common.SSHException'}
```

### PR DESCRIPTION
**Pull request by ylangisc**
Fixes #14196.

### CODE REVIEW

File: ssh/src/main/java/ch/cyberduck/core/sftp/SFTPSession.java [hunk 1]
```diff
@@ -278,19 +278,26 @@ private void authenticate(final SSHClient client, final Host host, final LoginCa
             final String identityAgent = new OpenSSHIdentityAgentConfigurator().getIdentityAgent(host.getHostname());
             switch(Factory.Platform.getDefault()) {
                 case windows:
```
Comment (dkocher) [inline ssh/src/main/java/ch/cyberduck/core/sftp/SFTPSession.java:280]: These checks are redundant it seems if proper checks for agent to be available in implementation for `getIdentities`

File: ssh/src/main/java/ch/cyberduck/core/sftp/auth/AgentAuthenticator.java [hunk 2]
```diff
@@ -25,4 +25,9 @@ public abstract class AgentAuthenticator {
     public abstract Collection<Identity> getIdentities();
 
     public abstract AgentProxy getProxy();
+
+    public boolean isAvailable() {
```
Comment (dkocher) [inline ssh/src/main/java/ch/cyberduck/core/sftp/auth/AgentAuthenticator.java:29]: Move this check instead to `getIdentities` of implementations.

**Review by dkocher — CHANGES_REQUESTED** [review]

**Review by dkocher — APPROVED** [review]

File: ssh/src/main/java/ch/cyberduck/core/sftp/openssh/OpenSSHAgentAuthenticator.java [hunk 3]
```diff
@@ -55,7 +55,11 @@ public AgentProxy getProxy() {
     @Override
     public Collection<Identity> getIdentities() {
         if(null == proxy) {
-            log.warn(String.format("SSH agent %s is not running", this));
+            log.warn("Missing proxy reference");
+            return Collections.emptyList();
+        }
+        if(!proxy.isRunning()) {
```
Comment (blaineam) [inline ssh/src/main/java/ch/cyberduck/core/sftp/openssh/OpenSSHAgentAuthenticator.java:61]: I think this additional conditional is breaking 1password for non IdentityFile uses. I don't think this check is needed for the original issue either.

**Review by blaineam — COMMENTED** [review]
This PR broke Mountain Duck for my 1Password 8 SSH setup. I'm pretty sure the section that broke it is the commented block.

**Pull request merged**